libraries/chain/abi_serializer.cpp

```diff
@@ -228,13 +237,35 @@ namespace eosio { namespace chain {
          return type;
    }
 
+   // true if std::string can be converted to name
+   bool abi_serializer::is_string_valid_name(std::string_view str) const
+   {
+      if (size_t len = str.size(); len > 13)
+         return false;
+
+      for( const char c : str ) {
+         if( c >= 'a' && c <= 'z' )
```

Comment:
check `name.hpp`
this is not full constraint. 13th character can't have same range. it needs special check.

Reply:
Also, shouldn't we already have a helper function somewhere that does this? I hope we're not reimplementing this logic all over the code.

Reply:
checked the 13th character to be one of ".12345abcdefghij" and added tests to name_tests.